guides/keras_cv/stl10_with_simclr.py

```diff
@@ -0,0 +1,205 @@
+"""
+Title: Contrastive Feature Learning with SimCLR in KerasCV
+Author: Ian Stenbit, [Luke Wood](https://lukewood.xyz)
+Date created: 2022/09/09
+Last modified: 2022/09/09
+Description: Use KerasCV for contrastive feature learning on STL-10
+"""
+
+"""
+## Overview
+
+In recent years, contrastive learning has enabled neural
+networks to learn from corpuses of unlabelled data.  Techniques such as SimCLR, SimSiam,
+and Masked AutoEncoders score within percentage points of state of the art techniques
+without leveraging labelled data.
+
+The STL-10 dataset is a benchmark dataset specifically targeted at self supervised
+learning techniques.  STL-10 consists of a large unlabelled data corpus and a tiny
+labelled corpus.  To solve the dataset, you typically use a contrastive learning
+technique to obtain a useful weight set, then finetune your model on the labelled data
+to obtain a classifier.
+
+KerasCV offers a contrastive learning API to perform self-supervised feature
+learning on unlabelled image data. In this guide, we'll demonstrate the use
+of this API to perform self-supervised feature learning and supervised fine-tuning
+on the STL-10 dataset.
+
+For a reference on contrastive learning, and SimCLR in particular, check out the
+[SimCLR paper](https://arxiv.org/pdf/2002.05709).
+
+To get started, let's sort out all of our imports and define configuration parameters.
+"""
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow import keras
+from tensorflow.keras import callbacks
+from tensorflow.keras import layers
+from tensorflow.keras import optimizers
+
+from keras_cv import losses
+from keras_cv import models
+from keras_cv import training
+
+
+CLASSES = 10
+BATCH_SIZE = 64
+IMAGE_SIZE = (96, 96)
+CONTRASTIVE_EPOCHS = 1
+FINE_TUNING_EPOCHS = 1
+
+"""
+## Data loading
+
+In this guide, we use the `stl10` dataset from tensorflow-datasets.
+
+This dataset includes 100,000 unlabelled training images, 5,000 labelled
+training images, and 8,000 labelled evaluation images.
+
+We start by loading the unlabelled examples. Note that we shuffle our unlabelled
+images, and specify `reshuffle_each_iteration` to ensure that training batches
+contain unique sets of images each epoch. This is important to maximize the
+effect of contrastive learning.
+"""
+
+unlabelled_images = tfds.load("stl10", split="unlabelled")
+unlabelled_images = unlabelled_images.map(
+    lambda entry: entry["image"], num_parallel_calls=tf.data.AUTOTUNE
+)
+
+unlabelled_images = unlabelled_images.shuffle(
+    buffer_size = 8*BATCH_SIZE, reshuffle_each_iteration=True
+)
+unlabelled_images = unlabelled_images.batch(BATCH_SIZE)
+
+"""
+We then load the labelled examples as separate train and test splits.
+"""
+
+train_labelled_images = tfds.load("stl10", split="train")
```

Comment:
Side note: we are not consistent across keras.io in how we spell "labeled/labelled". In general we tend to spell it "labeled" though